### Q1. Pearson correlation coefficient is a measure of the linear relationship between two variables. Suppose you have collected data on the amount of time students spend studying for an exam and their final exam scores. Calculate the Pearson correlation coefficient between these two variables and interpret the result.

In [1]:
import pandas as pd

df=pd.DataFrame({
    'study_time' : [10, 5, 8, 7, 6, 4, 9, 2, 3, 1],
'exam_scores' : [80, 65, 72, 68, 70, 60, 75, 50, 55, 45]
})

df.corr(method='pearson')

study_time  exam_scores
study_time     1.000000     0.981316
exam_scores    0.981316     1.000000

In [4]:
# correlation coefficient
df.corr().iloc[0, 1]

0.9813163270390649

The Pearson correlation coefficient is 0.98, which indicates a strong positive linear relationship between the two variables. This means that as the amount of time spent studying increases, the exam scores tend to increase as well. 

### Q2. Spearman's rank correlation is a measure of the monotonic relationship between two variables. Suppose you have collected data on the amount of sleep individuals get each night and their overall job satisfaction level on a scale of 1 to 10. Calculate the Spearman's rank correlation between these two variables and interpret the result.

In [7]:
df=pd.DataFrame({
    'sleep_time' : [8, 6, 7, 5, 9, 6, 8, 7, 7, 6],
'job_satisfaction' : [7, 5, 6, 4, 8, 5, 7, 6, 6, 5]
})

df.corr(method='spearman')

sleep_time  job_satisfaction
sleep_time               1.0               1.0
job_satisfaction         1.0               1.0

In [8]:
from scipy.stats import spearmanr

# example data
sleep_time = [8, 6, 7, 5, 9, 6, 8, 7, 7, 6]
job_satisfaction = [7, 5, 6, 4, 8, 5, 7, 6, 6, 5]

# calculate Spearman's rank correlation coefficient and p-value
corr, p_value = spearmanr(sleep_time, job_satisfaction)

# print the result
print("Spearman's rank correlation coefficient:", corr)
print("p-value:", p_value)


Spearman's rank correlation coefficient: 1.0
p-value: 0.0


The Spearman's rank correlation coefficient is 1.00 which indicates a strong monotonic relationship between the two variables. This means that as the amount of sleep individuals get each night increases, their overall job satisfaction tends to increase as well.

### Q3. Suppose you are conducting a study to examine the relationship between the number of hours of exercise per week and body mass index (BMI) in a sample of adults. You collected data on both variables for 50 participants. Calculate the Pearson correlation coefficient and the Spearman's rank correlation between these two variables and compare the results.

In [12]:
df=pd.DataFrame({
'exercise_hours' : [2.5, 3.0, 1.0, 2.0, 1.5, 4.0, 3.5, 2.0, 1.5, 2.5, 
                  2.0, 3.5, 2.5, 3.0, 1.5, 4.0, 2.5, 3.0, 1.0, 2.0, 
                  3.5, 2.0, 1.5, 3.0, 2.5, 2.0, 2.5, 1.0, 1.5, 3.0, 
                  2.0, 2.5, 3.0, 2.0, 1.5, 3.5, 1.0, 2.5, 3.0, 2.5, 
                  1.5, 3.5, 2.0, 2.5, 3.0, 1.5, 2.0, 3.5, 2.5, 2.0],
    
'bmi' : [25.1, 28.3, 32.4, 29.8, 31.7, 22.9, 26.5, 30.3, 29.5, 25.7, 
       28.8, 24.9, 25.4, 26.1, 31.2, 23.4, 25.8, 25.1, 31.9, 28.6, 
       25.5, 30.7, 30.8, 26.9, 27.5, 28.8, 25.6, 33.2, 29.5, 26.3, 
       31.9, 26.2, 27.1, 26.3, 30.7, 30.8, 24.9, 32.1, 25.8, 26.4, 
       31.4, 23.8, 29.6, 28.7, 26.5, 26.7, 31.6, 29.1, 27.9, 32.3]
})


In [13]:
# Pearson correlation coefficient
df.corr(method='pearson')

exercise_hours       bmi
exercise_hours        1.000000 -0.655075
bmi                  -0.655075  1.000000

In [16]:
print("Pearson correlation coefficient:",df.corr(method='pearson').iloc[0,1])

Pearson correlation coefficient: -0.6550752568429092


In [ ]:
# Spearman correlation coefficient
df.corr(method='pearson')

In [17]:
print("Spearman correlation coefficient:",df.corr(method='spearman').iloc[0,1])

Spearman correlation coefficient: -0.6323671411887234


- The Pearson correlation coefficient measures the strength and direction of a linear relationship between two variables. In this case, the Pearson correlation coefficient is -0.6550752568429092, indicating a weak, positive linear relationship between the number of hours of exercise per week and BMI. This means that as the number of hours of exercise per week increases, BMI also tends to increase slightly, but the relationship is not strong.

- the Spearman's rank correlation coefficient measures the strength and direction of a monotonic relationship between two variables, which means it measures how much one variable changes as the other variable changes, without assuming a specific functional form of the relationship. In this case, the Spearman's rank correlation coefficient is -0.6323671411887234, indicating a weak, positive monotonic relationship between the number of hours of exercise per week and BMI. This means that as the number of hours of exercise per week increases, BMI tends to increase slightly, but again, the relationship is not strong.

### Q4. A researcher is interested in examining the relationship between the number of hours individuals spend watching television per day and their level of physical activity. The researcher collected data on both variables from a sample of 50 participants. Calculate the Pearson correlation coefficient between these two variables.

In [27]:
import pandas as pd
import numpy as np

# Generate sample data for TV hours and physical activity
np.random.seed(42)
tv_hours = np.random.normal(3, 1, 50)
physical_activity = np.random.normal(5, 2, 50)

# Create a pandas DataFrame from the two arrays
df = pd.DataFrame({'TV Hours': tv_hours, 'Physical Activity': physical_activity})

a=df.corr(method='pearson').iloc[0,1]

print("Pearson correlation coefficient:",a)

Pearson correlation coefficient: 0.11007178534016061


### Q5. A survey was conducted to examine the relationship between age and preference for a particular brand of soft drink. The survey results are shown below:

In [18]:
df=pd.DataFrame({
    'Age':[25,42,37,19,31,28],
    'Soft Drink Preference':['Coke','Pepsi','Mountain Dew','Coke','Pepsi','Coke']
})
df

Age Soft Drink Preference
0   25                  Coke
1   42                 Pepsi
2   37          Mountain Dew
3   19                  Coke
4   31                 Pepsi
5   28                  Coke

In [31]:
a=df.corr(method='spearman').iloc[0,1]

In [32]:
print("Spearman Corrleation Coefficient:",a)

Spearman Corrleation Coefficient: 0.14237695078031212


### Q6. A company is interested in examining the relationship between the number of sales calls made per day and the number of sales made per week. The company collected data on both variables from a sample o  sales representatives. Calculate the Pearson correlation coefficient between these two variables.

In [33]:
import pandas as pd
import numpy as np

# Generate sample data for sales calls and sales made
np.random.seed(42)
sales_calls = np.random.normal(50, 10, 30)
sales_made = np.random.normal(10, 2, 30)

# Create a pandas DataFrame from the two arrays
df = pd.DataFrame({'Sales Calls': sales_calls, 'Sales Made': sales_made})

In [34]:
df.corr(method='pearson')

Sales Calls  Sales Made
Sales Calls     1.000000    0.098834
Sales Made      0.098834    1.000000

In [36]:
print("Pearson correlation coefficient:",df.corr(method='pearson').iloc[0,1])

Pearson correlation coefficient: 0.09883379170949537
